smithy-aws-iam-traits/src/main/resources/META-INF/smithy/aws.iam.smithy

```diff
@@ -64,6 +90,16 @@ list supportedPrincipalTypes {
     member: PrincipalType
 }
 
+/// A container for information on the resources that an IAM action may be authorized against.
+@private
+structure ActionResources {
+    /// Resources that will always be authorized against for functionality of the IAM action.
+    required: ActionResourceMap
+
+    /// Resources that will conditionally be authorized against for functionality of the IAM action.
+    optional: ActionResourceMap
```

Comment:
Nit: Are these conditional or optional? Im a little confused on what the authorization behavior for these resource will be. If they are conditional, what are they conditional on?